api/capi/src/nnstreamer-capi-util.c

```diff
@@ -898,33 +898,74 @@ ml_initialize_gstreamer (void)
 }
 
 /**
- * @brief Internal helper function to validate arguments
+ * @brief Internal helper function to validate model files.
  */
 static int
-_ml_validate_model_file (char **model, unsigned int num_models,
-    ml_nnfw_type_e * nnfw, gchar ** p1, gchar ** p2)
+_ml_validate_model_file (char **model, unsigned int num_models)
 {
-  if (!model) {
+  guint i;
+
+  if (!model || num_models < 1) {
     ml_loge ("The required param, model is not provided (null).");
     return ML_ERROR_INVALID_PARAMETER;
   }
-  if (num_models < 1 || !*model ||
-      !g_file_test (model[0], G_FILE_TEST_IS_REGULAR)) {
-    ml_loge ("The given param, model path [%s] is invalid or not given.",
-        GST_STR_NULL (model[0]));
-    return ML_ERROR_INVALID_PARAMETER;
+
+  for (i = 0; i < num_models; i++) {
+    if (!model[i] || !g_file_test (model[i], G_FILE_TEST_IS_REGULAR)) {
+      ml_loge ("The given param, model path [%s] is invalid or not given.",
+          GST_STR_NULL (model[i]));
+      return ML_ERROR_INVALID_PARAMETER;
+    }
   }
 
-  if (!nnfw)
+  return ML_ERROR_NONE;
+}
+
+/**
+ * @brief Gets available nn framework using file extension.
+ * @return @c 0 on success. Otherwise a negative error value.
+ */
+static int
+_ml_detect_framework (char **model, unsigned int num_models,
+    ml_nnfw_type_e * nnfw)
+{
+  int status = ML_ERROR_NONE;
+  gchar *detected;
+
+  detected =
+      gst_tensor_filter_framework_auto_detection ((const gchar **) model,
+      num_models);
+  if (detected == NULL) {
+    ml_loge ("The given model has unknown or not supported extension.");
     return ML_ERROR_INVALID_PARAMETER;
+  }
 
-  *p1 = g_ascii_strdown (model[0], -1);
-  if (num_models == 2)
-    *p2 = g_ascii_strdown (model[1], -1);
-  else if (num_models > 2)
-    return ML_ERROR_INVALID_PARAMETER; /** We do not have such a case, yet */
+  if (g_ascii_strcasecmp (detected, "nnfw") == 0) {
+    *nnfw = ML_NNFW_TYPE_NNFW;
+  } else if (g_ascii_strcasecmp (detected, "tensorflow-lite") == 0) {
+    *nnfw = ML_NNFW_TYPE_TENSORFLOW_LITE;
+  } else if (g_ascii_strcasecmp (detected, "vivante") == 0) {
+    *nnfw = ML_NNFW_TYPE_VIVANTE;
+  } else if (g_ascii_strcasecmp (detected, "tensorflow") == 0) {
+    *nnfw = ML_NNFW_TYPE_TENSORFLOW;
+  } else if (g_ascii_strcasecmp (detected, "custom") == 0) {
+    *nnfw = ML_NNFW_TYPE_CUSTOM_FILTER;
+  } else {
+    ml_loge ("The given model has unknown or not supported extension.");
+    status = ML_ERROR_INVALID_PARAMETER;
```

Comment:
gst_tensor_filter_framework_auto_detection() can returns other FWs such as caffe2, pytorch as so on.
In that case, does _ml_detect_framework() function return ML_ERROR_INVALID_PARAMETER?

Reply:
how about `ML_ERROR_NOT_SUPPORTED`?

Reply:
changed to not-supported